src/bindings/python/flux/wrapper.py

```diff
@@ -322,12 +331,15 @@ def __getattr__(self, name):
             return fun
 
         new_fun = self.check_wrap(fun, name)
-        new_method = six.create_bound_method(new_fun, weakref.proxy(self))
+        new_meth = MethodType(new_fun, self)
+
+        def wrap_class(self_renamed, *args, **kwargs):
+            return new_fun(self_renamed, *args, **kwargs)
```

Comment:
Nice workaround for making a callable class a method!  Can you throw a comment up above this to summarize why this is necessary for future reference?  Without the context of #2671, this seems like an unnecessary extra layer of indirection.

Reply:
Yes, yes I will.  Though I need to rework this as I made a kinda fundamental error that will keep this from working as we'd like.

Reply:
Added the comment, and re-worked wrapper inheritence.  Hopefully this will take care of it.  If not, it may be worth going cacheless for the time being until we can get to the generated binding phase.